rusk/src/lib/node/rusk.rs

```diff
@@ -150,6 +153,19 @@ impl Rusk {
                 continue;
             }
 
+            let tx_len = unspent_tx.size().unwrap_or_default();
+
+            if tx_len == 0 {
+                info!("Skipping {tx_id_hex} due to error while calculating the len");
+                continue;
+            }
+
+            if tx_len > size_left {
```

Comment:
it might be worth checking if we have no space for any more transactions (do we have a minimum tx size?), and if so stop, instead of trying all transactions.

Reply:
We could, but we have no minimum transaction size

Once introduced we can optimize the generator strategy as you suggested